chromium_src/net/base/url_util.cc

```diff
@@ -35,6 +35,10 @@ bool IsOnion(const GURL& url) {
   return url.DomainIs(kOnionDomain);
 }
 
+bool IsOnion(const url::Origin& origin) {
+  return origin.DomainIs(kOnionDomain);
```

Comment:
I think it might be better if `bool IsOnion(const GURL& url)` called `bool IsOnion(const url::Origin& origin)` passing the origin. It should be equivalent to what is here, but I think it's better when overloading to call the same code

Reply:
I didn't want to use `url::Origin::Create()` in `bool IsOnion(const GURL& url)` because of https://chromium.googlesource.com/chromium/src/+/main/docs/security/origin-vs-url.md and https://issues.chromium.org/issues/40805196, but maybe you were thinking of something else? Or for the purpose of making our override smaller, we should ignore those warnings?

Reply:
Settled on SchemeIsHTTPOrHTTPS || SchemeIsWSOrWSS after Slack discussion.